doc/modules/model_evaluation.rst

```diff
@@ -6,6 +6,113 @@
 Metrics and scoring: quantifying the quality of predictions
 ===========================================================
 
+.. _which_scoring_function:
+
+Which scoring function should I use?
+====================================
+
+Before diving into the details of the many scores and metrics, we want
+to give some guidance, inspired by statistical decision theory, on the choice
+of **scoring functions** for **supervised learning**, see [Gneiting2009]_:
+
+- *Which scoring function should I use?*
+- *Which scoring function is a good one for my task?*
+
+In a nutshell, if the scoring function is given, e.g. in a kaggle competition
+or in a business context, use that one.
+If you are free to choose, it starts by considering the ultimate goal and application
+of the prediction. It is useful to distinguish two steps:
+
+* Predicting
+* Decision making
+
+**Prediction:**
+Usually, the response variable :math:`Y` is a random variable, in the sense that there
+is *no deterministic* function :math:`Y = g(X)` of the features :math:`X`.
+Instead, there is a probability distribution :math:`F` of :math:`Y`.
+One can aim to predict the whole distribution, known as *probabilistic prediction*,
+or---more the focus of scikit-learn---issue a *point prediction* (or point forecast)
+by choosing a property or functional of that distribution :math:`F`.
+Typical examples are the mean (expected value), the median or a quantile of the
+response variable :math:`Y` (conditionally on :math:`X`).
+
+Once that is settled, use a **strictly consistent** scoring function for that
+target functional, see [Gneiting2009]_.
+For classification **strictly proper scoring rules**, see
+`wikipedia <https://en.wikipedia.org/wiki/Scoring_rule>`_ and [Gneiting2007]_, coincide
+with strictly consistent scoring functions.
+The table further below provides examples.
+One could say that consistent scoring functions act as *truth serum* in that
+they guarantee *"that truth telling [. . .] is an optimal strategy in
+expectation"* [Gneiting2014]_.
```

Comment:
Is this paragraph already talking about model evaluation? 
1. It seems so because of the term "scoring function" and its rules.
2. It doesn't seem so, since we are still under "Prediction" and also the "Decision Making" part below talks about how `y_pred` is made from `y_proba` in classifications, which is more related to a model's output than its evaluation.

Would it be better to add a new heading for this paragraph and perhaps move it to after the **Decision Making** section and right before **List of strictly consistent scoring functions:**?

Reply:
It really belongs to the "Prediction" paragraph. Note that a scoring function is best used for both:
- as loss function for model training
- as metric for model evaluation/comparison

Reply:
> Note that a scoring function is best used for both:
>     as loss function for model training
>     as metric for model evaluation/comparison

Oh yes! Thanks for adding this information above.
(This leads away, but now I wonder why `estimator.score() `doesn't get to use the scoring function used internally within fit().)

Reply:
> (This leads away, but now I wonder why estimator.score() doesn't get to use the scoring function used internally within fit().)

🤷 Wasn't my doing.